synapse/app/pusher.py

```diff
@@ -140,6 +167,26 @@ class PusherSlaveStore(
         StateStore.__dict__["_get_state_group_for_event"]
     )
 
+    _get_state_groups_from_groups = (
+        StateStore.__dict__["_get_state_groups_from_groups"]
+    )
+
+    _get_state_group_from_group = (
+        StateStore.__dict__["_get_state_group_from_group"]
+    )
+
+    get_global_account_data_by_type_for_users = (
+        AccountDataStore.__dict__["get_global_account_data_by_type_for_users"]
```

Comment:
I'm afraid this won't work. You'll need to implement cache invalidation for the account_data using the replication streams. There are examples of how to do that in ``synapse/replication/slave/storage``.